File: 프로그래머스/2/42842. 카펫/카펫.cs
using System;

public class Solution
{
    public int[] solution(int brown, int yellow)
    {
        int total = brown + yellow;

        for (int w = 3; w <= brown / 2; w++)  // 가로
        {
            for (int h = 3; h <= w; h++)      // 세로 (항상 w >= h)
            {
                int boundary = (w + h - 2) * 2;
                int center = w * h - boundary;

                if (brown == boundary && yellow == center)
                    return new int[] { w, h };
            }
        }

        return null; // 실제로는 도달하지 않음
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) 
{
    vector<int> answer;
    vector<pair<int,int>> couple;

    
    int cube = brown + yellow;
    
    for(int i=3;i<=cube;i++)
    {
        if((cube % i) == 0)
        {
            couple.push_back({i,(cube/i)}); // 가로,세로 
        }
    }
    

    
    for(int i=0;i<couple.size();i++)
    {
        int width = couple[i].first;
        int height = couple[i].second;
        
        if(width < height) continue;
        
        int guess = width * 2 + (height - 2) * 2;
        
        if(guess == brown) 
        {
            answer.push_back(width);
            answer.push_back(height);
            break;
        }
    }
    
    return answer;
}